lib/client.js

```diff
@@ -254,6 +338,16 @@ class AssetComputeClient extends EventEmitter {
         });
     }
 
+
+    async unregister() {
+        if (!this.assetCompute) {
+            await this.initialize();
+        }
+        const response = await this.assetCompute.unregister();
+        this.registered = false; // wait until successful unregister to set registered variable
```

Comment:
You should check the response is a case in which it actually unregistered, and not returned an error code. Because as far as I see, response is nearly the raw api response, which could be also a 404, a 500...

Reply:
I don't think that is necessary because the `register` method in `assetCompute` will actually throw if `!response.ok`:
https://github.com/adobe/asset-compute-client/blob/unregister/lib/assetcompute.js#L97-L99

Looking at the API, `!response.ok` for every error case except 404: https://git.corp.adobe.com/nui/nui/blob/master/doc/api.md#unregister
404 means there is no journal yet (no call to register yet) which would mean `this.registered = false` already, so resetting that will not change anything. So I don't think there is a strong need to look at the response text.

Reply:
Btw, the response is the response body, not the raw api

Reply:
Maybe consider it for later, this would make the code more robust, because we don't need to count on other code to not change. LGTM for now.